internal/provider/data_source_gitlab_project_issues_test.go

```diff
@@ -0,0 +1,69 @@
+package provider
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+func TestAccDataSourceGitlabProjectIssues_basic(t *testing.T) {
+	rInt := acctest.RandInt()
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: providerFactories,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDataGitlabProjectIssuesConfig(rInt),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("data.gitlab_project_issues.this", "issues.#", "2"),
+					resource.TestCheckResourceAttr("data.gitlab_project_issues.this", "issues.0.iid", "1"),
+					resource.TestCheckResourceAttr("data.gitlab_project_issues.this", "issues.0.title", "Issue 1"),
+					resource.TestCheckResourceAttr("data.gitlab_project_issues.this", "issues.0.description", "Description 1"),
+					resource.TestCheckResourceAttr("data.gitlab_project_issues.this", "issues.1.iid", "2"),
+					resource.TestCheckResourceAttr("data.gitlab_project_issues.this", "issues.1.title", "Issue 2"),
+					resource.TestCheckResourceAttr("data.gitlab_project_issues.this", "issues.1.description", "Description 2"),
+				),
+			},
+		},
+	})
+}
+
+func testAccDataGitlabProjectIssuesConfig(rInt int) string {
+	return fmt.Sprintf(`
```

Comment:
Point of information, and a bit of a nitpick for a data source, but, I would like to move in a direction where we are only including resources in the test TF config that are being tested.
- Makes tests faster
- Reduces issues where we cannot represent the "setup" behavior in TF config alone, providing good examples for future tests
- Allows us to use the `CheckDestroy` TestCase field correctly to check the destroy function, where we are destroying the tested resource without any of its dependencies like the project it lives in. (Not applicable to data sources, but still...)

So for instance here, the project and the two issues could be created in code before the call to `resource.Test...`. What do you think?

Reply:
Yes, absolutely, I'm also nitpicking this usually. It was for testing convenience to get something - and was one of the points I was planning do to when implementing more tests 👍